packages/core/src/Modules/Cashier/Containers/p2p-cashier.jsx

```diff
@@ -1,21 +1,63 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import queryString from 'query-string';
 import P2P from '@deriv/p2p';
 import { getLanguage } from '@deriv/translations';
 import { WS } from 'Services';
 import { connect } from 'Stores/connect';
 import ServerTime from '_common/base/server_time';
 
 /* P2P will use the same websocket connection as Deriv/Binary, we need to pass it as a prop */
-const P2PCashier = ({ currency, local_currency_config, is_virtual, residence, setNotificationCount }) => (
-    <P2P
-        websocket_api={WS}
-        lang={getLanguage()}
-        client={{ currency, local_currency_config, is_virtual, residence }}
-        server_time={ServerTime}
-        setNotificationCount={setNotificationCount}
-    />
-);
+class P2PCashier extends React.Component {
+    state = {
+        order_id: null,
+    };
+
+    setQueryOrder = input_order_id => {
+        const { history, location } = this.props;
+        const current_query_param = queryString.parse(this.props.location.search);
+        delete current_query_param.order;
+
+        const order_id_param = input_order_id
+            ? { ...current_query_param, order: input_order_id }
+            : { ...current_query_param };
+
+        if (this.state.order_id !== input_order_id) {
+            // Changing query params
+            history.push({
+                pathname: '/cashier/p2p',
```

Comment:
Is this compatible with Binary Static?

Reply:
thats why i put it outside of p2p package, then we will make this handler in binary static but not using history.push

Reply:
Woops I overlooked it was outside the package. Thanks 👍